src/robotFactory.js

```diff
@@ -1,10 +1,115 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    if ('x' in coords && 'y' in coords) {
+      this.coords = coords;
+    } else if ('x' in coords) {
+      this.coords = {
+        x: coords.x,
+        y: 0,
+      };
+    } else if ('y' in coords) {
+      this.coords = {
+        x: 0,
+        y: coords.y,
+      };
+    } else {
+      this.coords = {
+        x: 0,
+        y: 0,
+      };
+    }
```

Comment:
let's make your code smaller 
```suggestion
    this.coords = {
      x: coords.x || 0,
      y: coords.y || 0,
    };
```